### ISSUE TITLE
Help Request Procedure

### ISSUE THREAD

**Issue by FalseDev** [The-Coding-Academy/tca-helper-meet#4]
### My suggestion on new procedure to get help

In order to ask for help, one must follow this new procedure:
- Use a bot command in which they explain their problem/error in a help request channel
- The one asking will have to check that they've done a list of items
  - Learnt language basics
  - Googled the issue
  - Removed the token/other creds from any media/code they send
- It gets sent to a channel where anyone can *accept* this question and CodingBot v4 assigns a channel to both of them
- The person who accepts then helps resolve the issue
- The person then thanks the helper using `.thank`

**Comment by FrostByte266** [comment]
As long as it's not icing other people out from contributing to helping I think this is a good idea. The channel should remain publically visible, and then they will have to accept some kind of prompt to gain access. Maybe a slash command? :thinking:

**Comment by Iusedmy1stnamelol** [comment]
In order words like a ticket system but with commands. I agree but, can we switch to mod instead of Tickets for Moderation/Roles.

Another reason why the community figure is a good thing because in case helpers are offline. Community Figures has helped the server with coding ETC. They can also give support if a helper is offline or they are just there to give the support.
@FalseDev.

**Comment by Div-100** [comment]
> ### My suggestion on new procedure to get help
> In order to ask for help, one must follow this new procedure:
> 
> * Use a bot command in which they explain their problem/error in a help request channel
> * The one asking will have to check that they've done a list of items
>   
>   * Learnt language basics
>   * Googled the issue
>   * Removed the token/other creds from any media/code they send
> * It gets sent to a channel where anyone can _accept_ this question and CodingBot v4 assigns a channel to both of them
> * The person who accepts then helps resolve the issue
> * The person then thanks the helper using `.thank`

This solves all the issues imo
Lemme pin this issue

**Comment by FalseDev** [comment]
> As long as it's not icing other people out from contributing to helping I think this is a good idea. The channel should remain publically visible, and then they will have to accept some kind of prompt to gain access. Maybe a slash command? thinking

Yep, I meant the help requests channel will be publicly visible so anyone can *accept* a help request

**Comment by BobDotCom** [comment]
I've actually been planning something like this for coding bot. I had planned something like the python server help channels, and the code is even open source so I could work on that a bit. Can someone make an issue on the coding bot repo?

**Comment by Tanishquasar** [comment]
@BobDotCom Such feature is planned for TechStruck as well, might be implemented soon enough.
